mycore-base/src/main/java/org/mycore/tools/MCRTopologicalSort.java

```diff
@@ -104,8 +105,25 @@ public void prepareData(String[] files, File dir) {
                             if (xmlStreamReader.getLocalName().equals("mycoreobject")) {
                                 nodes.forcePut(i, xmlStreamReader.getAttributeValue(null, "ID"));
                             } else if (xmlStreamReader.getLocalName().equals("parent")) {
-                                parentNames.put(i,
+                                List<String> dependecyList;
+                                if(parentNames.containsKey(i)){
```

Comment:
`dependencyList = parentNames.computeIfAbsent(i, i -> new ArrayList())` is much shorter than using the complicated if-then-else